lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -885,23 +881,28 @@ for DIRECTLY_CHAINED dependencies:
 sub duplicate {
     my ($self, $args) = @_;
     $args ||= {};
-    my $schema = $self->result_source->schema;
 
     # If the job already has a clone, none is created
-    return unless $self->can_be_duplicated;
+    my ($orig_id, $clone_id) = ($self->id, $self->clone_id);
+    return "Job $orig_id is still scheduled"                   if $self->state eq SCHEDULED;
+    return "Job $orig_id has already been cloned as $clone_id" if defined $clone_id;
 
     my $jobs = $self->cluster_jobs(skip_parents => $args->{skip_parents}, skip_children => $args->{skip_children});
-    log_debug("Jobs to duplicate " . dump($jobs));
-    try {
-        $schema->txn_do(sub { $self->_create_clones($jobs, $args->{prio}, $args->{skip_ok_result_children}); });
-    }
-    catch {
-        my $error = shift;
-        log_debug("rollback duplicate: $error");
-        die "Rollback failed during failed job cloning!"
-          if ($error =~ /Rollback failed/);
-        $jobs = undef;
+    log_debug('Duplicating jobs: ' . dump($jobs));
+    eval {
```

Comment:
Idly wondering why you're moving away from the `try/catch` which I thought we wanted to use more...that said, your error handling looks rock solid.

Reply:
I did it in this particular case because I wanted to return from the function from within the catch block. That's unfortunately not possible with `Try::Tiny`. Besides these details, `eval + if (my $error = $@)` is equivalent to `Try::Tiny` in my opinion and I have no preference which to use.

Reply:
Right, that makes sense, I hadn't thought of that